DSharpPlus.Commands/Exceptions/ChecksFailedException.cs

```diff
@@ -1,11 +1,9 @@
 namespace DSharpPlus.Commands.Exceptions;
 
-using System;
 using System.Collections.Generic;
-using DSharpPlus.Commands.ContextChecks;
 
-public sealed class ChecksFailedException(IReadOnlyDictionary<ContextCheckAttribute, Exception> errors, string commandName, string? message = null)
+public sealed class ChecksFailedException(IReadOnlyList<string> errors, string commandName, string? message = null)
     : CommandsException(message ?? $"Checks for {commandName} failed.")
 {
-    public IReadOnlyDictionary<ContextCheckAttribute, Exception> FailingContextChecks { get; init; } = errors;
+    public IReadOnlyList<string> Errors { get; init; } = errors;
```

Comment:
`IReadOnlyDictionary<ContextCheckAttribute, string?>` for more context available to the user.

Reply:
What do you mean by "cleanly search the dictionary"? If you're still wanting your proposed behavior, then the user could iterate over the dictionary's values.

Also a list of tuples... </3. That just removes the key aspect of the dictionary. To get that behavior the user could iterate over the dictionary as is.

Reply:
it's pretty hard to key this dictionary anyway, since these are reference types following reference type semantics